File: app-android/app/src/main/java/com/aengix/hopper/model/HopNodeProfile.kt
package com.aengix.hopper.model

import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class HopNodeProfile(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val host: String = "",
    val port: Int = HopConstants.DEFAULT_SSH_PORT,
    val user: String = "",
    val privateKey: String = "",
    val hostKeys: List<String> = emptyList(),
    val installDir: String = "",
) {
    val trimmedName: String get() = name.trim()
    val trimmedHost: String get() = host.trim()
    val trimmedUser: String get() = user.trim()

    val displayName: String
        get() = when {
            trimmedName.isNotEmpty() -> trimmedName
            trimmedHost.isNotEmpty() -> trimmedHost
            else -> "Untitled"
        }

    val resolvedInstallDir: String
        get() = installDir.trim().ifEmpty { HopConstants.DEFAULT_INSTALL_DIR }
}

File: app/Shared/HopNodeProfile.swift
import Foundation

struct HopNodeProfile: Codable, Equatable, Identifiable, Hashable {
    var id: UUID = UUID()
    var name: String = ""
    var host: String = ""
    var port: Int = HopConstants.defaultSSHPort
    var user: String = ""
    var privateKey: String = ""
    var hostKeys: [String] = []
    /// Server bundle path from QR v2 (`install_dir`), e.g. `~/hopper`.
    var installDir: String = ""
    /// Hopper server bundle version from deploy/QR v2 (`server_version`).
    var serverVersion: String = ""
    /// Minimum app version required by the server (`min_app_version`).
    var minAppVersion: String = ""

    var trimmedName: String {
        name.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    var trimmedHost: String {
        host.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    var trimmedUser: String {
        user.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    var displayName: String {
        if !trimmedName.isEmpty { return trimmedName }
        if !trimmedHost.isEmpty { return trimmedHost }
        return "Untitled"
    }

}
